packages/vaadin-overlay/src/vaadin-overlay.js

```diff
@@ -559,6 +559,8 @@ class OverlayElement extends ThemableMixin(DirMixin(ControllerMixin(PolymerEleme
         this.dispatchEvent(evt);
       });
 
+      document.addEventListener('keydown', this._boundKeydownListener);
```

Comment:
The modeless dialogs should probably only close on Esc when focused. An example with a text field outside the dialog whose value is cleared using Esc:

https://user-images.githubusercontent.com/1222264/168827089-74e22a90-4d89-4f26-b525-6f4e00f6ed3a.mp4

Reply:
There's a similar issue if you have, for example, a date picker inside the dialog and close its overlay with Esc:

https://user-images.githubusercontent.com/1222264/168829161-fc501c5a-7fbb-4f59-b327-ff3f9e8ec15f.mp4

Reply:
> a date picker inside the dialog and close its overlay with Esc

a similar issue with clicking outside was fixed last year https://github.com/vaadin/web-components/issues/2978

Reply:
Created a new issue for closing date-picker inside a dialog: #3871

Reply:
@tomivirkki Added logic to detect if the overlay contains focus based on the `event.composedPath()`.

Reply:
That works. If you have the focus outside the modeless dialog, it seems to be difficult to get the focus back inside it though. It's not enough to just click the text inside the dialog, but it needs to be some focusable element. Not sure if this is a problem or not.

https://user-images.githubusercontent.com/1222264/169095402-8d6345e9-1455-41da-9086-62f5ba2cf177.mp4

Reply:
I also noticed that. Maybe we could update logic related to `mousedown` to detect whether overlay got focus back.